show-portal-user-info/src/main/java/com/esri/arcgismaps/sample/showportaluserinfo/components/AuthenticationAppViewModel.kt

```diff
@@ -0,0 +1,125 @@
+
+    val authenticatorState: AuthenticatorState = AuthenticatorState()
+
+    // create a ViewModel to handle dialog interactions as a public val
+    val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
+
+    private val noPortalInfoText = application.getString(R.string.no_portal_info)
+    private val startInfoText = application.getString(R.string.start_info_text)
+    private val arcGISUrl = "https://www.arcgis.com"
+    private val oAuthUserConfiguration = OAuthUserConfiguration(
+        arcGISUrl,
+        // This client ID is for sample purposes only. For use of the Authenticator in your own app,
+        // create your own client ID. For more info see: https://developers.arcgis.com/documentation/mapping-apis-and-services/security/tutorials/register-your-application/
+        "rR5K4TEyhkPDcTkf",
+        "my-ags-app://auth"
+    )
+
+    private val _portalUserName = MutableStateFlow(String())
+    val portalUserName: StateFlow<String> = _portalUserName.asStateFlow()
+
+    private val _emailID = MutableStateFlow(String())
+    val emailID: StateFlow<String> = _emailID.asStateFlow()
+
+    private val _userCreationDate = MutableStateFlow(String())
+    val userCreationDate: StateFlow<String> = _userCreationDate.asStateFlow()
+
+    private val _portalName = MutableStateFlow(String())
+    val portalName: StateFlow<String> = _portalName.asStateFlow()
+
+    private val defaultBitmap = BitmapFactory.decodeResource(application.resources, R.drawable.user)
+
+    private val _userThumbnail: MutableStateFlow<Bitmap> = MutableStateFlow(defaultBitmap)
+    val userThumbnail: StateFlow<Bitmap> = _userThumbnail.asStateFlow()
+
+    private val _infoText: MutableStateFlow<String> = MutableStateFlow(startInfoText)
+    val infoText: StateFlow<String> = _infoText.asStateFlow()
+
+    private val _isLoading: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val isLoading: StateFlow<Boolean> = _isLoading.asStateFlow()
+
+    private val _url: MutableStateFlow<String> = MutableStateFlow(arcGISUrl)
+    val url: StateFlow<String> = _url.asStateFlow()
+    fun setUrl(newUrl: String) {
+        _url.value = newUrl
+    }
+
+    fun signOut() = viewModelScope.launch {
+        _isLoading.value = true
+        ArcGISEnvironment.authenticationManager.signOut()
+        _infoText.value = startInfoText
+        _isLoading.value = false
+        _portalUserName.value = ""
+        _emailID.value = ""
+        _userCreationDate.value = ""
+        _portalName.value = ""
+        _userThumbnail.value = defaultBitmap
+    }
+
+    fun loadPortal() = viewModelScope.launch {
+        _isLoading.value = true
+        authenticatorState.oAuthUserConfiguration = oAuthUserConfiguration
+        val portal = Portal(url.value, Portal.Connection.Authenticated)
+        portal.load().also {
+            _isLoading.value = false
+        }.onFailure {
+            messageDialogVM.showMessageDialog("Failed to load portal", it.message.toString())
```

Comment:
@gunt0001 I had [that implementation initially](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/133#discussion_r1322049912) where I show the error message in the infoText section of the app. 
I updated to use the MessageDialog based on @shubham7109 's suggestion on the PR. 

I don't have strong opinion to use either or. I am fine with both. 
Since, we did have a MessageDialog composable component in the common samples lib directory, thought it would be good to make use of it in our samples, which enhances the sample UI design a bit.